src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -166,6 +166,28 @@ protected BaseDiagnosticIncrementalAnalyzer(DiagnosticAnalyzerService owner, Wor
         public abstract Task<IEnumerable<DiagnosticData>> GetDiagnosticsForSpanAsync(Document document, TextSpan range, CancellationToken cancellationToken);
         #endregion
 
+        #region build error synchronization
+        /// <summary>
+        /// Callback from build listener.
+        /// 
+        /// Given diagnostics are errors host got from explicit build.
+        /// It is up to each analyzer how they will merge this information with live diagnostic info.
```

Comment:
Minor: sorry for being pedantic, but can we say "incremental analyzer" instead of "analyzer" in the comment. Confuses me very often between these two.